lib/mutant.rb

```diff
@@ -193,22 +192,22 @@ class Config
 
     DEFAULT = new(
       debug:             false,
-      fail_fast:         false,
-      integration:       Integration::Null,
-      matcher:           Matcher::Config::DEFAULT,
-      includes:          EMPTY_ARRAY,
-      requires:          EMPTY_ARRAY,
-      isolation:         Mutant::Isolation::Fork,
-      reporter:          Reporter::CLI.build($stdout),
-      zombie:            false,
-      jobs:              Mutant.ci? ? CI_DEFAULT_PROCESSOR_COUNT : ::Parallel.processor_count,
       expected_coverage: Rational(1),
       expression_parser: Expression::Parser.new([
         Expression::Method,
         Expression::Methods,
         Expression::Namespace::Exact,
         Expression::Namespace::Recursive
-      ])
+      ]),
+      fail_fast:         false,
+      includes:          EMPTY_ARRAY,
+      integration:       Integration::Null,
+      isolation:         Mutant::Isolation::Fork,
+      jobs:              Mutant.ci? ? CI_DEFAULT_PROCESSOR_COUNT : ::Parallel.processor_count,
```

Comment:
We discussed this in the past and the fact you do not recall it tells us we should have an exmplaining comment here.

CI systems run mostly in containerized environments where the host has many more CPUs available than the containers itself can utilize, enforced by resource limits. As there is no standard way to detect the resource limits of a container, and only the hosts resources can be read: Mutant would spin up many more threads above the container quota, trashing and eventually be killed by the CI host.

We simply choose the default of 4 reflecting our usual CircleCI projects. I wounder if we should remove this CI specific default and let everyon figure out by himself the correct jobs setting, only defaulting to the detected CPI count?

As this code is part of the diff, but was not semantically changed, I do not consider this a blocker. I made a dedicated issue to continue the discussion. https://github.com/mbj/mutant/issues/501